src/Control/Carrier/NonDet/Church.hs

```diff
@@ -58,7 +66,7 @@ instance Alternative (NonDetC m) where
 
 instance Monad (NonDetC m) where
   NonDetC a >>= f = NonDetC $ \ fork leaf nil ->
-    a fork (\ a' -> runNonDetC (f a') fork leaf nil) nil
+    a fork (runNonDet fork leaf nil . f) nil
```

Comment:
This was by no means the point of this PR, but it’s certainly a nice side benefit that we can use the eliminators point-free.